neutron_plugin/port.py

```diff
@@ -166,6 +168,18 @@ def connect_security_group(neutron_client, **kwargs):
     sgs = port['security_groups'] + [security_group_id]
     neutron_client.update_port(port_id, {'port': {'security_groups': sgs}})
 
+    # Double check if SG has been actually updated (a race-condition
```

Comment:
im not sure this actually handles the problem correctly
rather than an openstack bug, this actually seems like a racing condition in our code - note that unlike in the Server-SG connection, where the API is along the lines of "server.addSG(SG)", here the API is "port.update(SGs=sgs)", i.e. we redefine all of the port's SGs, rather than adding a new one
this means that if two such operations are run in parallel, each might try to add one port without taking into consideration the other one.
the new protection code will ensure the port is indeed connected to the given SG at any one operation, but there could be a scenario where even with this code, only one of two such SGs which get connected in parallel actually does end up on the port.